src/robot/libraries/Process.py

```diff
@@ -524,8 +526,37 @@ def _manage_process_timeout(self, handle, on_timeout):
 
     def _wait(self, process):
         result = self._results[process]
-        result.rc = process.wait() or 0
-        result.close_streams()
+
+        _take_stdout = (process.stdout and (not process.stdout.closed))
+        _take_stderr = (process.stderr and (not process.stderr.closed))
+
```

Comment:
Style:
- We don't generally use empty line inside methods. If methods get so long that they need to be grouped by empty lines, we prefer introducing helper methods. In this particular case empty lines could just be removed.
- We don't use `_` prefix with local variables.